src/cpu/x64/injectors/jit_uni_binary_injector.cpp

```diff
@@ -2647,22 +2647,57 @@ template <typename T>
 typename std::enable_if<!(std::is_same<T, Xbyak::Zmm>::value
                           || std::is_same<T, Xbyak::Address>::value)>::type
 jit_uni_binary_injector_t<isa, Vmm>::execute_prelu_binary(const Vmm &dst,
-                                                        const Vmm &lhs, const T &rhs) const {
-    // in sse4 vmm_aux0 as mask it's index must be 0
-    Vmm vmm_aux0 = Vmm(rhs_arg_static_params_.rhs_prelu_helper_vmm_idx);
-
-    if (dst == vmm_aux0) {
-        vmm_aux0 = Vmm(14);
-        if (isa == sse41)
-            assert(!"conflict mask register");
-    }
+    const Vmm &lhs, const T &rhs) const {
+    if (is_superset(isa, avx)) {
+        host_->uni_vmulps(rhs, rhs, lhs);
+        host_->uni_vblendvps(dst, lhs, rhs, lhs);
+    } else {
+        using dnnl::impl::utils::one_of;
+        // in sse4 vmm_aux0 as mask it's index must be 0
+        Vmm vmm_aux0 = Vmm(rhs_arg_static_params_.rhs_prelu_helper_vmm_idx);
+
+        if (one_of(vmm_aux0, dst, lhs, rhs)) {
+            //let's find a vacant XMM register
+            int occupied_idices[] = {dst.getIdx(), lhs.getIdx(), rhs.isMEM() ? -1 : rhs.getIdx()};
 
-    push_vmm(host_, vmm_aux0);
-    host_->uni_vmulps(rhs, rhs, lhs);
-    host_->vpxor(vmm_aux0, vmm_aux0, vmm_aux0);
-    host_->vcmpltps(vmm_aux0, lhs, vmm_aux0);
-    host_->uni_vblendvps(dst, lhs, rhs, vmm_aux0);
-    pop_vmm(host_, vmm_aux0);
+            int fixup_reg_indx = 14;
+            while (std::any_of(std::begin(occupied_idices), std::end(occupied_idices),
+                [&](const int x) {return x == fixup_reg_indx;}) && --fixup_reg_indx > 0) {}
+            if (fixup_reg_indx < 0) assert(!"couldn't find a vacant XMM reg");
+
+            vmm_aux0 = Vmm(fixup_reg_indx);
+        }
+
+        const auto aux_orig_indx = vmm_aux0.getIdx();
+
+        push_vmm(host_, vmm_aux0);
+
+        if (aux_orig_indx != 0) {
+            // XMM0 is occupied, let's swap registers
+            int occupied_idices[] = {dst.getIdx(), lhs.getIdx(), rhs.isMEM() ? -1 : rhs.getIdx()};
+            for (const int indx : occupied_idices) {
+                if (indx == 0) {
+                    auto vmm_zero = Vmm(indx);
+                    host_->vmovups(vmm_aux0, vmm_zero);
+                    std::swap(vmm_aux0, vmm_zero);
```

Comment:
Seems have some problem here.Assuming:
 vmm_aux =vmm(4) ,  vmm_zero =vmm(0),  lhs = vmm(0) before swapping. After swap, vmm_aux=vmm(0), vmm_zero=vmm(4), but lhs still vmm(0).  Seem lhs should also be changed to vmm(4).  right?

Reply:
Good catch. Will fix it

Reply:
Done